### PR TITLE
Start working towards R5 version, with the feedback obtained so far.

### PR DESCRIPTION
**Pull request by lucteo**
- streamline the introduction
- move the examples section after introduction
- updated Design section with the latest code changes, and updated wording
- streamline the discussions on replaceability, and include feedback from Ms and Contracts paper
- streamline the "API for implementing custom system contexts" section
- change conclusion in "Need for the `system_context` class" section
- remove "Questions to ask LEWG/SG1/vendors"
- minor other updates

### CODE REVIEW

File: paper_framework_sources/p2079_system_execution_context.bs [hunk 1]
```diff
@@ -58,154 +65,309 @@ This was removed from [[P2300R9]] based on that and other input.
...
+An application might have multiple parts, possibly in different binaries; different parts of the application may not know of each other.
+Thus, different parts of the application might use different parallel engines.
+This can create several problems:
+* oversubscription because of different thread pools
+* problems with nested parallel loops (one parallel loop is called from the other)
+* problems related to interaction between different parallel engines
+* other performance problems
+* etc.
+
+To solve these problems we propse a shared parallel execution context that:
```
Comment (LeeHowes) [inline paper_framework_sources/p2079_system_execution_context.bs:78]: Typo -> propose

File: paper_framework_sources/p2079_system_execution_context.bs [hunk 2]
```diff
@@ -504,25 +574,18 @@ Underneath the `system_context` there is a singleton of some sort, potentially o
...
-
-By making the context explicit we require users to drain their work before they drain the context.
-In debug builds, at least, we can also add reference counting so that destruction of the context before work completes reports a clear error to ensure that people clean up.
-That is harder to do if the context is destroyed at some point after main completes.
-This lifetime question also applies to construction: we can lazily construct a thread pool before we first use a scheduler to it.
-
-For this reason, and consistency with other discussions about structured concurrency, we opt for an explicit context object here.
+In R4 and previous revisions, we opted for the explicit context object.
+The argument was that providing explicit contexts is easier to reason about the lifetime of the schedulers.
+However, reasoning for the lifetime of schedulers (and the operations spawned on them) may not be that useful.
```
Comment (LeeHowes) [inline paper_framework_sources/p2079_system_execution_context.bs:583]: This is a really weak statement.

File: paper_framework_sources/p2079_system_execution_context.bs [hunk 3]
```diff
@@ -504,25 +574,18 @@ Underneath the `system_context` there is a singleton of some sort, potentially o
...
+Ultimatelly, all the operations launched on schedulers need to be contained in the lifetime of `main()`.
```
Comment (LeeHowes) [inline paper_framework_sources/p2079_system_execution_context.bs:584]: Typo. 

Statement is fine, assuming we can define "main" broadly enough.

File: paper_framework_sources/p2079_system_execution_context.bs [hunk 4]
```diff
@@ -504,25 +574,18 @@ Underneath the `system_context` there is a singleton of some sort, potentially o
...
+Adding an artificial scope for reasoning about lifetimes is just an extra burden.
```
Comment (LeeHowes) [inline paper_framework_sources/p2079_system_execution_context.bs:585]: People are likely to have concerns about very visible singleton-like nature of this thing. We can call get_system_scheduler() from anywhere and it's going to be extremely easy to slip up on lifetimes. Is it reasonable to expect runtimes to do lifetime checking? Can we even in production builds ensure that tasks error out after main?

File: paper_framework_sources/p2079_system_execution_context.bs [hunk 5]
```diff
@@ -58,154 +65,309 @@ This was removed from [[P2300R9]] based on that and other input.
...
-  ...
+  template&lt;receiver R>
+  requires receiver_of<R>
+  <i>impl-defined-operation_state</i> connect(R&&) && noexcept(std::is_nothrow_constructible_v&lt;std::remove_cvref_t&lt;R>, R>);
 };
-```
+</pre>
+
+ - `get_system_scheduler()` returns a scheduler that provides a view on some underlying execution context supporting *parallel forward progress*, with at least one thread of execution (which may be the main thread).
+ - two objects returned by `get_system_scheduler()` will share the same execution context.
```
Comment (lucteo) [inline paper_framework_sources/p2079_system_execution_context.bs:268]: TODO: relax this statement to allow run-time replaceability

File: paper_framework_sources/p2079_system_execution_context.bs [hunk 6]
```diff
@@ -58,79 +65,171 @@ This was removed from [[P2300R9]] based on that and other input.
...
-The `system_scheduler` instance does not have to outlive work submitted to it.
-The `system_scheduler` is technically implementation-defined, but must be nameable.
-See later discussion for how this might work.
+The `system_scheduler` customises `bulk`, so we can use `bulk` dependent on the scheduler.
+Here we use it in structured form using the parameterless `get_scheduler` that retrieves the scheduler from the receiver, combined with `on`:
+```cpp
+auto bar() {
+  return
+    ex::let_value(
+      ex::get_scheduler(),          // Fetch scheduler from receiver.
```
Comment (rarutyun) [inline paper_framework_sources/p2079_system_execution_context.bs:149]: Seems like this is outdated usage or get_scheduler() according to what we discussed on the sync up

File: paper_framework_sources/p2079_system_execution_context.bs [hunk 7]
```diff
@@ -316,57 +403,37 @@ The paper considers compile-time replaceability as not being a valid option beca
...
-
-One of the potential mitigations of run-time replaceability issues (such as data races) is that we could say that <em>"the
-`system_scheduler` shall be replaced only once and before any outstanding work submitted to it by the application in `main`"</em> or something similar. It not a formal wording, of course; it's an example to give some ruminations.
+Replaceability is also part of the [[P2900R7]] proposal for the contract-violation handler.
+The paper proposes that whether the handler is replaceable to be implementation-defined.
+If an implementation chooses to support replaceability, it shall be done similar to replacing the global `operator new` and `operator delete` (link-time replaceability).
+
+The feedback we received from Microsoft, is that they are not interested in supporting replaceability on their platforms.
+They would prefer that we offer implementations an option to not implement replaceabiilty.
+Moreover, for systems for which replaceability is supported they would prefer to make the replaceabiilty mechanism to be implementation defined.
```
Comment (rarutyun) [inline paper_framework_sources/p2079_system_execution_context.bs:412]:
```suggestion
Moreover, for systems where replaceability is supported they would prefer to make the replaceabiilty mechanism to be implementation defined.
```

**Pull request merged**